File: 9.回文数.js
/*
 * @lc app=leetcode.cn id=9 lang=javascript
 *
 * [9] 回文数
 */

// @lc code=start
/**
 * @param {number} x
 * @return {boolean}
 */
var temp = 5;
var isPalindrome = function(x) {
    if(x < 0){
        return false;
    }
    let xTemp = x,remain,base,res = 0;
    while(xTemp != 0){
        base = xTemp % 10;
        remain = parseInt(xTemp/10);
        res = res*10 + base;
        xTemp = remain;
    }
    if(res == x){
        return true;
    }
    return false;
};
// @lc code=end
let arr = 121
res = isPalindrome(arr);
console.log(res);

File: 9.回文数.py
#
# @lc app=leetcode.cn id=9 lang=python3
#
# [9] 回文数
#

# @lc code=start
from typing import SupportsAbs


class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        resStr = 0
        soureNum = x
        while x > 0:
            base = x % 10 
            temp = x//10
            resStr = resStr*10 + base
            x = temp
        if resStr == soureNum:
            return True
        return False
# @lc code=end
s = Solution()
res = s.isPalindrome(121)